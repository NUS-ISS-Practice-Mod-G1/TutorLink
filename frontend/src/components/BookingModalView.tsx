import React from "react";

interface BookingModalViewProps {
  booking: {
    studentName: string;
    tutorName: string;
    date: Date;
    slot: { start: string; end: string };
    lessonType: string;
  };
  onClose: () => void;
}

const BookingModalView: React.FC<BookingModalViewProps> = ({
  booking,
  onClose,
}) => {
  return (
    <div className="fixed inset-0 flex items-center justify-center bg-black bg-opacity-40 z-50">
      <div className="bg-white rounded-xl shadow-lg p-6 w-96">
        <h2 className="text-lg font-semibold mb-4 text-gray-800">
          Booking Details
        </h2>

        {booking.studentName && (
          <p className="mb-1">👤 Student: {booking.studentName}</p>
        )}
        {booking.tutorName && (
          <p className="mb-1">👨‍🏫 Tutor: {booking.tutorName}</p>
        )}

        <p className="mb-1">
          📅 {booking.date.toDateString()} | ⏰ {booking.slot.start} -{" "}
          {booking.slot.end}
        </p>
        <p className="mb-1">🎓 Lesson Type: {booking.lessonType}</p>
        <div className="flex justify-end mt-4">
          <button
            onClick={onClose}
            className="px-4 py-2 rounded-md bg-gray-200 hover:bg-gray-300"
          >
            Close
          </button>
        </div>
      </div>
    </div>
  );
};

export default BookingModalView;
