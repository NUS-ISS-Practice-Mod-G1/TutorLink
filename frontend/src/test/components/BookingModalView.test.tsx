import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import BookingModalView from "@/components/BookingModalView";

describe("BookingModalView", () => {
  const mockOnClose = jest.fn();

  const booking = {
    studentName: "Alice Tan",
    tutorName: "John Tutor",
    date: new Date("2025-11-10T10:00:00Z"),
    slot: { start: "10:00", end: "11:00" },
    lessonType: "Mathematics",
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  // ---------- BASIC RENDER ----------
  test("renders modal with all booking details", () => {
    render(<BookingModalView booking={booking} onClose={mockOnClose} />);

    // Header
    expect(screen.getByText("Booking Details")).toBeInTheDocument();

    // Student & Tutor
    expect(screen.getByText(/Student: Alice Tan/)).toBeInTheDocument();
    expect(screen.getByText(/Tutor: John Tutor/)).toBeInTheDocument();

    // Date & time
    expect(screen.getByText(/10:00 - 11:00/)).toBeInTheDocument();
    expect(screen.getByText(/Nov/)).toBeInTheDocument(); // formatted month

    // Lesson type
    expect(screen.getByText(/Mathematics/)).toBeInTheDocument();
  });

  // ---------- CONDITIONAL RENDER ----------
  test("does not render student or tutor lines when not provided", () => {
    const partialBooking = {
      ...booking,
      studentName: "",
      tutorName: "",
    };

    render(<BookingModalView booking={partialBooking} onClose={mockOnClose} />);

    expect(screen.queryByText(/Student:/)).not.toBeInTheDocument();
    expect(screen.queryByText(/Tutor:/)).not.toBeInTheDocument();
  });

  // ---------- CLOSE BUTTON ----------
  test("calls onClose when Close button clicked", () => {
    render(<BookingModalView booking={booking} onClose={mockOnClose} />);
    fireEvent.click(screen.getByText("Close"));
    expect(mockOnClose).toHaveBeenCalledTimes(1);
  });

  // ---------- SNAPSHOT ----------
  test("matches snapshot", () => {
    const { container } = render(<BookingModalView booking={booking} onClose={mockOnClose} />);
    expect(container).toMatchSnapshot();
  });
});
