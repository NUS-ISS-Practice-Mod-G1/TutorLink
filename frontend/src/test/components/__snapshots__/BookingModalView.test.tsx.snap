// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BookingModalView matches snapshot 1`] = `
<div>
  <div
    class="fixed inset-0 flex items-center justify-center bg-black bg-opacity-40 z-50"
  >
    <div
      class="bg-white rounded-xl shadow-lg p-6 w-96"
    >
      <h2
        class="text-lg font-semibold mb-4 text-gray-800"
      >
        Booking Details
      </h2>
      <p
        class="mb-1"
      >
        👤 Student: 
        Alice Tan
      </p>
      <p
        class="mb-1"
      >
        👨‍🏫 Tutor: 
        John Tutor
      </p>
      <p
        class="mb-1"
      >
        📅 
        Mon Nov 10 2025
         | ⏰ 
        10:00
         -
         
        11:00
      </p>
      <p
        class="mb-1"
      >
        🎓 Lesson Type: 
        Mathematics
      </p>
      <div
        class="flex justify-end mt-4"
      >
        <button
          class="px-4 py-2 rounded-md bg-gray-200 hover:bg-gray-300"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
